examples/aws-sagemaker/endpoint.py

```diff
@@ -0,0 +1,219 @@
+            }
+        ]
+        self.endpoint_config_name = config_name
+        self.endpoint_config = {
+            "EndpointConfigName": self.endpoint_config_name,
+            "ProductionVariants": self.production_variants,
+        }
+        self.endpoint_name = endpoint_name
+
+    def create_image(self):
+
+        subprocess.run(self.build_cmd, shell=True)
+
+    def create_ecr_repo(self):
+
+        try:
+            ecr = boto3.client("ecr", region_name=self.region_name)
+            ecr.create_repository(repositoryName=self.repo_name)
+
+        except ecr.exceptions.RepositoryAlreadyExistsException:
+            pass
+
+        repo_check = ecr.describe_repositories(repositoryNames=[self.repo_name])
+        pprint(repo_check["repositories"])
+
+    def push_image(self):
+
+        subprocess.call(["sh", self.push_script])
+
+    def create_model(self):
+
+        self.sm_boto3.create_model(
+            ModelName=self.model_name,
+            Containers=self.image,
+            ExecutionRoleArn=self.ROLE_ARN,
+            EnableNetworkIsolation=False,
+        )
+
+    def create_endpoint_config(self):
+
+        self.sm_boto3.create_endpoint_config(**self.endpoint_config)
+
+    def create_endpoint(self):
+
+        self.sm_boto3.create_endpoint(
+            EndpointName=self.endpoint_name,
+            EndpointConfigName=self.endpoint_config_name,
+        )
+
+        pprint(self.sm_boto3.describe_endpoint(EndpointName=self.endpoint_name))
+
+    def destroy_endpoint(self):
+
+        self.sm_boto3.delete_endpoint(EndpointName=self.endpoint_name)
+        self.sm_boto3.delete_endpoint_config(
+            EndpointConfigName=self.endpoint_config_name
+        )
+        self.sm_boto3.delete_model(ModelName=self.model_name)
+
+        print("endpoint and SageMaker model deleted")
+
+
+def parse_args():
+
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        dest="action",
+        type=str,
+        choices=["create", "destroy"],
+        help="Args to 'create' or 'destroy' a SageMaker endpoint.",
+    )
+
+    return parser.parse_args()
+
+
+def main():
+
+    """
+    process arguments and initialize SparseMaker object
+    """
+
+    args = parse_args()
+
+    SM = SparseMaker(
+        instance_count=1,
+        region_name="us-east-1",
+        instance_type="ml.c5.2xlarge",
+        image_name="deepsparse-sagemaker-example",
+        repo_name="deepsparse-sagemaker",
+        image_push_script="./push_image.sh",
+        variant_name="QuestionAnsweringDeepSparseDemo",
+        model_name="question-answering-example",
+        config_name="QuestionAnsweringExampleConfig",
+        endpoint_name="question-answering-example-endpoint",
+        role_arn="<PLACEHOLDER>",
+    )
+
+    if args.action == "create":
+        SM.create_image()
+        SM.create_ecr_repo()
+        SM.push_image()
+        SM.create_model()
+        SM.create_endpoint_config()
+        SM.create_endpoint()
+
+    else:
```

Comment:
generally, it's not great to default to destroy without checking that that's what the user passed in. This can lead to bugs in the future if we were to add more options and a later implementor forgets to update here. It'd be better if we could migrate back to the original code that was marked as resolved here. As noted above, though, moving to click will take care of issues like this for us so we don't have to maintain the code

Reply:
@markurtz it isn't defaulting to destroy, the argparse `action` param is required and errors if you don't provide one of the choices. you have to make a choice on the CLI and click gives the same functionality in this case. as said before, we should wait on click as a refactor to all of the examples here

Reply:
Correct, but if a new choice is added in the future this opens up a potential pathway that a later implementor may forget about. Generally, it's never good to default to an else statement and the code across SparseML, SparseZoo, and DeepSparse has generally adhered to this principle to make future maintenance and testing easier. Towards the inclusion of click, it's a simple change that removes this issue altogether

Reply:
if someone adds a new choice they have to add to `choices=["create", "destroy"],`

Reply:
Correct, but again the if else condition becomes a weak point especially if there are multiple of them. This is a general design pattern and code maintence we've strived to keep across the codebase -- no default else statements in the event that there are pathways that people forget to update. If the next implementor updates the choices but forgets to update one of the if else statements that is called only in certain conditions, we're now deleting in that case rather than raising an exception.

Reply:
Same for click, new pathways added into the code base were agreed upon to go through click rather than argparse because it limits the amount of code and testing we have to own.